python/cudf/cudf/core/dataframe.py

```diff
@@ -796,6 +796,8 @@ def __str__(self):
         return self.to_string()
 
     def astype(self, dtype, errors="raise", **kwargs):
+        if self._num_columns == 0:
```

Comment:
Pandas return updated column with expected dtype, but this seems to return same set of columns.

Reply:
This is in case of no columns, could you share an example snippet?

Reply:
Apologies, I got confused it with empty rows.

Reply:
Sure no worries. Actually changed that piece of code now.